google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/HeaderInterceptor.java

```diff
@@ -0,0 +1,129 @@
+    return new SimpleForwardingClientCall<ReqT, RespT>(next.newCall(method, callOptions)) {
+      @Override
+      public void start(Listener<RespT> responseListener, Metadata headers) {
+        TagContext tagContext = getTagContext(headers, method.getFullMethodName());
+        super.start(
+            new SimpleForwardingClientCallListener<RespT>(responseListener) {
+              @Override
+              public void onHeaders(Metadata metadata) {
+                processHeader(metadata, tagContext);
+                super.onHeaders(metadata);
+              }
+            },
+            headers);
+      }
+    };
+  }
+
+  private void processHeader(Metadata metadata, TagContext tagContext) {
+    MeasureMap measureMap = STATS_RECORDER.newMeasureMap();
+    if (metadata.get(SERVER_TIMING_HEADER_KEY) != null) {
+      String serverTiming = metadata.get(SERVER_TIMING_HEADER_KEY);
+      Matcher matcher = SERVER_TIMING_HEADER_PATTERN.matcher(serverTiming);
+      if (matcher.find()) {
+        try {
+          long latency = Long.parseLong(matcher.group("dur"));
+          measureMap.put(SPANNER_GFE_LATENCY, latency).record(tagContext);
+          measureMap.put(SPANNER_GFE_HEADER_MISSING_COUNT, 0L).record(tagContext);
+        } catch (NumberFormatException e) {
+          logger.log(level, "invalid server-timing object in header");
+        }
+      }
+    } else {
+      measureMap.put(SPANNER_GFE_HEADER_MISSING_COUNT, 1L).record(tagContext);
+    }
+  }
+
+  private TagContext getTagContext(
+      String method, String projectId, String instanceId, String databaseId) {
+    return tagger
+        .currentBuilder()
+        .putLocal(SpannerRpcViews.PROJECT_ID, TagValue.create(projectId))
+        .putLocal(INSTANCE_ID, TagValue.create(instanceId))
+        .putLocal(DATABASE_ID, TagValue.create(databaseId))
+        .putLocal(SpannerRpcViews.METHOD, TagValue.create(method))
+        .build();
+  }
+
+  private TagContext getTagContext(String method) {
+    return tagger
+        .currentBuilder()
+        .putLocal(PROJECT_ID, TagValue.create("undefined-project"))
+        .putLocal(INSTANCE_ID, TagValue.create("undefined-instance"))
+        .putLocal(DATABASE_ID, TagValue.create("undefined-database"))
+        .putLocal(SpannerRpcViews.METHOD, TagValue.create(method))
+        .build();
+  }
+
+  private TagContext getTagContext(Metadata headers, String method) {
+    if (headers.get(GOOGLE_CLOUD_RESOURCE_PREFIX_KEY) != null) {
+      String googleResourcePrefix = headers.get(GOOGLE_CLOUD_RESOURCE_PREFIX_KEY);
+      try {
+        DatabaseId database = DatabaseId.of(googleResourcePrefix);
+        String databaseId = database.getDatabase();
+        String instanceId = database.getInstanceId().getInstance();
+        String projectId = database.getInstanceId().getProject();
+        return getTagContext(method, projectId, instanceId, databaseId);
+      } catch (IllegalArgumentException e) {
+        logger.log(level, e.toString());
```

Comment:
Could you log the whole exception with a message as well?

Maybe `logger.log(level, "Error parsing google cloud resource header", e)`